nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-shell-authorizer/src/main/java/org/apache/nifi/authorization/util/ShellRunner.java

```diff
@@ -33,13 +33,18 @@
 
     static String SHELL = "sh";
     static String OPTS = "-c";
-    static Integer TIMEOUT = 60;
 
-    public static List<String> runShell(String command) throws IOException {
+    private final long timeoutSeconds;
+
+    public ShellRunner(final long timeoutSeconds) {
```

Comment:
Does `timeoutSeconds` need to be a `long`? `Integer.MAX_VALUE = 2,147,483,647`, which converted from seconds is ~25,000 days.

Reply:
It definitely doesn't in terms of the size of the value, but since the value is obtained from this which returns a long:
`Math.round(FormatUtils.getPreciseTimeDuration(propertyValue, TimeUnit.SECONDS))`
and then passed in here which accepts a long:
`boolean waitFor(long timeout, TimeUnit unit)`
We could still convert to int for the internal variables if you prefer.

Reply:
I'm going to truncate to an int just for cleanliness. Thanks.